File: api/integration-execution-contract/src/main/java/com/integration/execution/contract/rest/response/kaseware/KwDocField.java
package com.integration.execution.contract.rest.response.kaseware;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class KwDocField {
    private int id;           // sequential id starting at 1
    private String fieldName; // GraphQL field name
    private String fieldType; // ofType.name or fallback to type.name
    private Boolean isMandatory; // true if non-nullable
}

File: web/src/api/models/KwDocField.ts
/**
 * Response DTO for source field mapping data
 * Represents field metadata from source system
 */
export interface KwDocField {
  /** Sequential ID starting at 1 */
  id: number;

  /** Field name from source system */
  fieldName: string;

  /** Field type (data type) */
  fieldType: string;

  /** Whether the field is mandatory/required */
  isMandatory?: boolean;
}
